src/shogun/neuralnets/NeuralNetwork.cpp

```diff
@@ -572,12 +573,12 @@ float64_t CNeuralNetwork::check_gradients(float64_t approx_epsilon, float64_t s)
 	SGMatrix<float64_t> y(get_num_outputs(),1);
 
 	for (int32_t i=0; i<x.num_rows; i++)
-		x[i] = CMath::random(0.0,1.0);
+		x[i] = random::random(0.0,1.0, m_prng);
 
 	// the outputs are set up in the form of a probability distribution (in case
 	// that is required by the output layer, i.e softmax)
 	for (int32_t i=0; i<y.num_rows; i++)
-		y[i] = CMath::random(0.0,1.0);
+		y[i] = random::random(0.0,1.0, m_prng);
```

Comment:
inefficient... and actually fill_array would be better...

Reply:
@vigsterkr  no more random::random stuff.
All the code base is ported now, I will start porting unit tests, and will come back to the code base afterwards for revision